test-integ/topoutil/asserter.go

```diff
@@ -439,3 +439,115 @@ func (a *Asserter) TokenExist(t *testing.T, cluster string, node *topology.Node,
+				if err != nil {
+					r.Log("err at node", node.Name, err)
+					continue
+				}
+
+				// Got the Autopilot server health response, break the loop
+				r.Log("Got response at node", node.Name)
+				break
+			}
+		}
+		r.Log("out", out, "health", out.Healthy)
+		require.Equal(r, expectedHealthy, out.Healthy)
+	})
+	return
+}
+
+type AuditEntry struct {
+	CreatedAt time.Time `json:"created_at"`
+	EventType string    `json:"event_type"`
+	Payload   Payload   `json:"payload"`
+}
+
+type Payload struct {
+	ID        string    `json:"id"`
+	Version   string    `json:"version"`
+	Type      string    `json:"type"`
+	Timestamp time.Time `json:"timestamp"`
+	Auth      Auth      `json:"auth"`
+	Request   Request   `json:"request"`
+	Response  Response  `json:"response,omitempty"` // Response is not present in OperationStart log
+	Stage     string    `json:"stage"`
+}
+
+type Auth struct {
+	AccessorID  string    `json:"accessor_id"`
+	Description string    `json:"description"`
+	CreateTime  time.Time `json:"create_time"`
+}
+
+type Request struct {
+	Operation   string            `json:"operation"`
+	Endpoint    string            `json:"endpoint"`
+	RemoteAddr  string            `json:"remote_addr"`
+	UserAgent   string            `json:"user_agent"`
+	Host        string            `json:"host"`
+	QueryParams map[string]string `json:"query_params,omitempty"` // QueryParams might not be present in all cases
+}
+
+type Response struct {
+	Status string `json:"status"`
+	Error  string `json:"error,omitempty"` // Error field is optional,shows up only with the status is non 2xx
+}
+
+func (a *Asserter) ValidateHealthEndpointAuditLog(t *testing.T, filePath string) {
+	boolIsErrorJSON := false
+	jsonData, err := os.ReadFile(filePath)
+	require.NoError(t, err)
+
+	// Print details of each AuditEntry
+	var entry AuditEntry
+	events := strings.Split(strings.TrimSpace(string(jsonData)), "\n")
+
+	// function to validate the error object when the health endpoint is returning 429
+	isErrorJSONObject := func(errorString string) bool {
+		// Attempt to unmarshal the error string into a map[string]interface{}
+		var m map[string]interface{}
+		err := json.Unmarshal([]byte(errorString), &m)
+		if err != nil {
+			// Error occurred during unmarshaling, indicating the string is not a JSON object
+			return false
+		}
+		return true
+	}
+
+	for _, event := range events {
+		if strings.Contains(event, "/v1/operator/autopilot/health") {
+			if strings.Contains(event, "OperationComplete") {
+				err = json.Unmarshal([]byte(event), &entry)
+				require.NoError(t, err, "Error unmarshalling JSON: %v\", err")
+				if entry.Payload.Response.Status == "429" {
+					fmt.Printf("Payload Response Error: %s\n", entry.Payload.Response.Error)
+					errResponse := entry.Payload.Response.Error
+					if isErrorJSONObject(errResponse) {
+						fmt.Println("Autopilot Health endpoint error response in the audit log looks fine")
```

Comment:
are these debugging statements that were left in? we should be able to remove these then and get rid of the `else` branch here